File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/api/OrderApi.java
package ru.elenacherev.librarymanager.api;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ru.elenacherev.librarymanager.api.dto.Order;

public interface OrderApi {

	Order getOrder(
			@NotNull Long orderId
	);

	Page<Order> searchOrders(
			Pageable pageable
	);

	Order createOrder(
			@Valid Order order
	);

	Order saveOrder(
			@NotNull Long orderId, 
			@Valid Order order
	);

}

File: librarymanager-api/src/main/java/ru/elenacherev/librarymanager/api/OrderApi.kt
package ru.elenacherev.librarymanager.api

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import ru.elenacherev.librarymanager.api.dto.Order
import java.util.*
import javax.validation.Valid

interface OrderApi {

    fun getOrder(
        orderId: UUID
    ): Order?

    fun searchOrders(
        pageable: Pageable
    ): Page<Order>

    fun createOrder(
        order: @Valid Order
    ): Order

    fun saveOrder(
        orderId: UUID,
        order: @Valid Order
    ): Order
}